src/Model/FileSupport/BuyRequest/BundleDataProvider.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/wishlist-graphql
+ * @link    https://github.com/scandipwa/wishlist-graphql
+ */
+
+declare(strict_types=1);
+
+namespace ScandiPWA\WishlistGraphQl\Model\FileSupport\BuyRequest;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Wishlist\Model\Wishlist\BuyRequest\BuyRequestDataProviderInterface;
+use Magento\Wishlist\Model\Wishlist\Data\WishlistItem;
+
+/**
+ * Data provider for bundle product buy requests
+ */
+class BundleDataProvider implements BuyRequestDataProviderInterface
+{
+    private const PROVIDER_OPTION_TYPE = 'bundle';
+
+    /**
+     * @inheritdoc
+     *
+     * @phpcs:disable Magento2.Functions.DiscouragedFunction
+     */
+    public function execute(WishlistItem $wishlistItem, ?int $productId): array
```

Comment:
The 2 _foreach_ loops look super similar, besides the entry data and validation.
Maybe there is a way to bring this out in a function and call that twice to reduce the amount of code duplication?